src/cmds/hardware/lsusb/lsusb.c

```diff
@@ -28,18 +31,104 @@ static void show_usb_dev(struct usb_dev *usb_dev) {
 			usb_dev->dev_desc.id_vendor,
 			usb_dev->dev_desc.id_product);
 
+
+}
+
+static void show_usb_desc_device(struct usb_dev *usb_dev) {
+	printf(" Device Descriptor:\n"
+			"   b_length             %5u\n"
+			"   b_desc_type          %5u\n"
+			"   bcd_usb              %2x.%02x\n"
+			"   b_dev_class          %5u\n"
+			"   b_dev_subclass       %5u\n"
+			"   b_dev_protocol       %5u\n"
+			"   b_max_packet_size    %5u\n"
+			"   id_vendor           0x%04x\n"
+			"   id_product          0x%04x \n"
+			"   bcd_device           %2x.%02x\n"
+			"   i_manufacter         %5u\n"
+			"   i_product            %5u\n"
+			"   i_serial_number      %5u\n"
+			"   i_num_configurations %5u\n\n",
+			usb_dev->dev_desc.b_length, 
+			usb_dev->dev_desc.b_desc_type,
+			usb_dev->dev_desc.bcd_usb >> 8, usb_dev->dev_desc.bcd_usb & 0xff,
+			usb_dev->dev_desc.b_dev_class,
+			usb_dev->dev_desc.b_dev_subclass,
+			usb_dev->dev_desc.b_dev_protocol, 
+			usb_dev->dev_desc.b_max_packet_size,
+			usb_dev->dev_desc.id_vendor, 
+			usb_dev->dev_desc.id_product,
+			usb_dev->dev_desc.bcd_device >> 8, usb_dev->dev_desc.bcd_device & 0xff,
+			usb_dev->dev_desc.i_manufacter,
+			usb_dev->dev_desc.i_product,
+			usb_dev->dev_desc.i_serial_number,
+			usb_dev->dev_desc.i_num_configurations);
 }
 
+static void show_usb_desc_interface(struct usb_dev *usb_dev) {
+	printf(" Interface Descriptor:\n"
+			"   b_length             %5u\n"
+			"   b_desc_type          %5u\n"
+			"   b_interface_number   %5u\n"
+			"   b_alternate_setting  %5u\n"
+			"   b_num_endpoints      %5u\n"
+			"   b_interface_class    %5u\n"
+			"   b_interface_subclass %5u\n"
+			"   b_interface_protocol %5u\n"
+			"   i_interface          %5u\n",
+			usb_dev->iface_desc.b_length, 
+			usb_dev->iface_desc.b_desc_type, 
+			usb_dev->iface_desc.b_interface_number,
+			usb_dev->iface_desc.b_alternate_setting, 
+			usb_dev->iface_desc.b_num_endpoints, 
+			usb_dev->iface_desc.b_interface_class,
+			usb_dev->iface_desc.b_interface_subclass,
+			usb_dev->iface_desc.b_interface_protocol,
+			usb_dev->iface_desc.i_interface);
+}
+
+static void show_usb_desc_configuration(struct usb_dev *usb_dev) {
+	struct usb_desc_configuration *config = \
+				(struct usb_desc_configuration *) usb_dev->config_buf;
+	
+	if(!config) {
+		return;
+	}
+
+	printf(" Configuration Descriptor:\n"
+			"   b_length             %5u\n"
+			"   b_desc_type          %5u\n"
+			"   w_total_length      0x%04x\n"
+			"   b_num_interfaces     %5u\n"
+			"   bConfigurationValue  %5u\n"
+			"   i_configuration      %5u\n"
+			"   bm_attributes         0x%02x\n"
+			"   b_max_power          %5u\n\n",
+			config->b_length, 
+			config->b_desc_type,
+			config->w_total_length,
+			config->b_num_interfaces, 
+			config->b_configuration_value,
+			config->i_configuration,
+			config->bm_attributes,
+			config->b_max_power);	
+}
+
+
 int main(int argc, char **argv) {
 	struct usb_dev *usb_dev = NULL;
-	int opt;
+	int opt, flag = 0;
```

Comment:
btw the variable name `flag` is not very good, it's better to use `flag_verbose` at least..